flang/runtime/descriptor.cpp

```diff
@@ -162,11 +162,17 @@ RT_API_ATTRS int Descriptor::Allocate() {
     elementBytes = raw_.elem_len = 0;
   }
   std::size_t byteSize{Elements() * elementBytes};
+
+  // Force default allocator in device code.
+#ifdef RT_DEVICE_COMPILATION
```

Comment:
This logic could be encapsulated in a single member function in Descriptor and used both here and below.

Reply:
Yeah it can actually be done directly in GetAllocIdx. I'll update that.

Reply:
GetAllocIdx is the only way to decode that field, and that's its purpose.  Wrap it with something new to do the replacement for these purposes.  It doesn't have to be in Descriptor if it only needs to be used in this source file.